### PR TITLE
feat(CreateTearsheet/FullPage): #1538 add support for dynamic steps

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #1538 

This PR allows for the usage of dynamic steps in the CreateTearsheet and CreateFullPage components. The introduction of the `includeStep` prop for the CreateTearsheetStep and CreateFullPageStep components is how you would utilize the dynamic step approach. I have included example usage in the mdx doc files for CreateTearsheet and CreateFullPage.

#### What did you change?
```
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.mdx
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.stories.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.test.js
packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPageStep.js
packages/cloud-cognitive/src/components/CreateInfluencer/CreateInfluencer.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.test.js
packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js
packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepTearsheet.js
packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepWithIntro.js
packages/cloud-cognitive/src/global/js/hooks/useCreateComponentStepChange.js
packages/cloud-cognitive/src/global/js/hooks/useResetCreateComponent.js
```
#### How did you test and verify your work?
Storybook and verified that the tests for these two components pass.

Thanks for the help on this one @dcwarwick 🎉 

Note: I opted out of using `.reduce` to get the `firstIncludedStep` and `lastIncludedStep` because some of the `prev` items were coming back as `undefined` so I found another way that worked.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js [hunk 1]
```diff
@@ -64,11 +70,33 @@ export let CreateTearsheet = forwardRef(
...
+    const lastIndexOf = (array, key) => {
+      for (let i = array.length - 1; i >= 0; i--) {
+        if (array[i]?.[key] === true) {
+          return i + 1;
+        }
+      }
+      return -1;
+    };
+
+    useEffect(() => {
```
Comment (davidmenendez) [inline packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js:88]: if this is the same code used in `CreateFullPage` you should definitely make them a shared utility

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js [hunk 2]
```diff
@@ -22,68 +22,93 @@ export let CreateTearsheetStep = forwardRef(
...
     // title, secondaryLabel, and introStep props so that it can be sent to the CreateInfluencer.
     useEffect(() => {
-      const stepHasReported = steps?.stepData?.includes(
-        (item) => item.title === title
-      );
-      if (!stepHasReported && steps?.stepData?.length < steps?.totalStepCount) {
-        steps?.setStepData((prev) => [
-          ...prev,
-          {
+      if (stepsContext) {
```
Comment (davidmenendez) [inline packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js:66]: same comment about reusing code

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks good for the most part. as always with this particularly complex component i'm assuming it's all in working order. i don't see any glaring issues.

my only real comment is that it looks like you're using the same code in a few spots. considering how complex this component is we should really aim to reuse code to cut down on bloat and improve efficiency and readability.

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js (hunk 2 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js]: Updated, this part here as well as the same part in `CreateFullPageStep` are now using a custom hook called `useRetrieveStepData`.

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js (hunk 1 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.js:80]: Created utility called `lastIndexOf.js`

**Review by davidmenendez — APPROVED** [review]
thanks for making those changes. looks good to me!

**Pull request merged**